server/src/instance_snapshot/patch_compute.rs

```diff
@@ -0,0 +1,147 @@
+use std::collections::{HashMap, HashSet};
+
+use rbx_dom_weak::{RbxTree, RbxId, RbxInstance};
+
+use super::{
+    snapshot::InstanceSnapshot,
+    patch::{PatchSet, PatchChildren, PatchChildrenEntry, PatchProperties},
+};
+
+pub fn compute_patch<'a>(
+    snapshot: &'a InstanceSnapshot,
+    tree: &RbxTree,
+    id: RbxId,
+    patch_set: &mut PatchSet<'a>,
+) {
+    let instance = tree.get_instance(id)
+        .expect("Instance did not exist in tree");
+
+    if instance.class_name != snapshot.class_name {
+        panic!("NYI: changing class name of an instance?");
```

Comment:
NYI = Not Yet Implemented, so why not use `unimplemented!("changing class name of an instance?")`

Reply:
I didn't know `unimplemented!` accepted a string argument, good idea!